cmd/workspace/workspace/overrides.go

```diff
@@ -19,7 +27,36 @@ func exportOverride(exportCmd *cobra.Command, exportReq *workspace.ExportRequest
 	exportCmd.Annotations["template"] = `{{.Content | b64_decode}}`
 }
 
+func importOverride(importCmd *cobra.Command, importReq *workspace.Import) {
+	var filePath string
+	importCmd.Use = "import TARGET_PATH"
+	importCmd.Flags().StringVar(&filePath, "file", "", `Path of local file to import`)
+	importCmd.MarkFlagsMutuallyExclusive("content", "file")
+
+	originalRunE := importCmd.RunE
+	importCmd.RunE = func(cmd *cobra.Command, args []string) error {
+		if filePath != "" {
+			b, err := os.ReadFile(filePath)
+			if err != nil {
+				return err
+			}
+			importReq.Content = base64.StdEncoding.EncodeToString(b)
+		}
+		err := originalRunE(cmd, args)
+		target := &apierr.APIError{}
+		if errors.As(err, &target) &&
+			(importReq.Format == workspace.ImportFormatSource || importReq.Format == "") &&
+			target.StatusCode == http.StatusBadRequest &&
+			strings.Contains(target.Message, "The zip file may not be valid or may be an unsupported version.") {
+			return fmt.Errorf("%w Hint: Objects imported using source format are assumed to be zip encoded by default. Please specify a language using the --language flag if you are trying to import a notebook instead", err)
```

Comment:
When exactly are objects expected to be zip encoded?

The check for this error can/should be factored into a separate function.

It's hard to read when this error is wrapped.

Reply:
Objects are expected to zip encoded when `fomat=source` or `format=""`. Source format is the API default.

Refactored wrapping the error into a separate function.

Reply:
We expect the ZIP to be either a single compressed source notebook or a compressed directory full of source notebooks.

We error if it's there's a file present that is not a source format notebook. For example even jupyter notebooks would trigger this error.